qlib/fileinfo.rs

```diff
@@ -241,6 +252,27 @@ impl FdWaitInfo {
         return Ok(());
     }
 
+    pub fn UpdateFDSync(&self, fd: i32) -> Result<()> {
+        let mask = {
+            let fi = self.lock();
+
+            let mask = fi.queue.Events();
+            if mask == fi.mask {
+                return Ok(());
+            }
+
+            mask
+        };
+
+        return Self::waitfd(fd, mask);
```

Comment:
Can't the body of the function be called directly here?

Reply:
sorry that I don't understand the question. where can the function body called directly?

Reply:
What I meant is, why not calling
```
WaitFDAsync(...)
```
directly?

Reply:
I still doesn't understand. WaitFDAsync needs call UpdateFDSync. If we can WaitFDAsync in UpdateFDSync, it is recursive call.

Reply:
I don't know if we are talking about the same thing :). What I see here is the call-chain:
```
fn UpdateFDSync(...)
       |
      ...
       |
  Self::waitfd(...)   ------>  fn waitfd(x,y)
       |                          |  
       |                      HostSpace::WaitFDAsync(x,y) --------> ....
       |                          |
       |             <------    Ok(()) 
      ///
```
Because `fn waitfd(...)` does not apply any logic to the arguments, I would guess that it is equivalent to:
```
fn UpdateFDSync(...)
       |
      ...
       |
  HostSpace::WaitFDAsync(x,y) --------> ....
       | 
    Ok(())
       |
      ///
```
except from the additional stack layout that gets created by the function call. That is why I asked.

* [fn waitfd(...)](https://github.com/QuarkContainer/Quark/blob/e75568ce4123e9ed854d09ef284ccfbc46fb3754/qlib/fileinfo.rs#L270)
* [fn WaitFDAsync(...)](https://github.com/QuarkContainer/Quark/blob/e75568ce4123e9ed854d09ef284ccfbc46fb3754/qlib/kernel/Kernel.rs#L937)